src/java/main/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -430,6 +432,7 @@ public void processRequest(Request request) {
             // the client and leader disagree on where the client is most
             // recently attached (and therefore invalid SESSION MOVED generated)
             cnxn.sendCloseSession();
+            request.checkLatency();
```

Comment:
It was intended to measure end to end latency of a request. Hence latency check after the request is completed.